zarr/core.py

```diff
@@ -1879,13 +1899,21 @@ def _process_chunk(
                         else dim
                         for i, dim in enumerate(self.chunks)
                     ]
-                    cdata.read_part(start, nitems)
-                    chunk_partial = self._decode_chunk(
-                        cdata.buff,
-                        start=start,
-                        nitems=nitems,
-                        expected_shape=expected_shape,
-                    )
+                    if isinstance(cdata, UncompressedPartialReadBufferV3):
```

Comment:
isinstance checks all around the place are unfortunate and make it hard to make new codecs that do something related but different. How would we make this a protocol?

Reply:
That's indeed a very unfortunate spaghetti-code :spaghetti: , I simply took over the design of the previous partial reads, but it's just becoming too complex with more options. I'd love to do a proper refactoring of those code-paths, but not in this PR, to keep them reviewable.  I think it would help to have some of the logic in the compressor itself, but I don't have sth. specific planned out.

Reply:
@jstriebel: can you capture the outstanding refactorings in an issue (if you don't already have work prepped in a PR... 🙂 )

Reply:
Added issue #1329 for it